### ISSUE TITLE
FATAL: Failed to start the monitor thread.

### ISSUE THREAD

**Issue by ateiss** [dimitri/pgloader#1190]
I get the following error output:

FATAL: Failed to start the monitor thread.
The value 619777536 is not of type FIXNUM.

I'm using pgloder 3.6.2
What's going wrong here?

**Comment by phoe** [comment]
What is your implementation and CPU architecture?

**Comment by ateiss** [comment]
Linux MA-1200 4.4.45-c2pv #6 SMP Fri Mar 10 14:52:32 CET 2017 x86_64 GNU/Linux

**Comment by phoe** [comment]
Which Lisp implementation are you using? Is it a 64-bit one?

**Comment by ateiss** [comment]
pgloader is an 32-Bit executable. Is this the Problem?

**Comment by phoe** [comment]
Yes - it seems that your 32-bit Lisp implementation runs into type problems, since a 30-bit number like `619777536` doesn't fit into a tagged integer, called `fixnum` in the Lisp world.

Could you try rebuilding pgloader with a 64-bit Lisp implementation and then retry?

@dimitri - does it make sense for you to support 32-bit Lisp implementations? If yes, then this is a bug in pgloader.

**Comment by ateiss** [comment]
With the 64-Bit build of pgloader it works.
Thanks.

**Comment by phoe** [comment]
Thanks for confirmation.

Leaving this issue open since I don't know if 32-bit builds are supported.

**Comment by dimitri** [comment]
I don't think it's interesting for us to support 32-bit builds, but @df7cb might say otherwise, as we have some 32-bits platform in the Debian package builds. I'll defer the decision here to @df7cb ; unless the fix is too complex to maintain.

**Comment by phoe** [comment]
The fix is not complex to maintain, I think - you can use reader conditionals for the type, something like `#+32-bit (unsigned-byte 64) #+64-bit fixnum #-(or 32-bit 64-bit) (error "...")`

**Comment by dimitri** [comment]
I don't know where in the code you're hinting we do that... care to prepare a patch or a PR? ;-)

**Comment by phoe** [comment]
> I don't know where in the code you're hinting we do that...

We'll need to get a backtrace from @ateiss - then we'll know.

**Comment by df7cb** [comment]
Hi,
pgloader was stalled in Debian and apt.postgresql.org for the past two years because the 32-bit support was broken. Now it works again (the problem was in some library, probably cl+ssl), so it would be very sad to see it gone again. 32-bit is old and smells, but there is no reason yet to ditch it completely, so I'd appreciate it if we put some effort in to keep it alive. Thanks.

**Comment by phoe** [comment]
Thanks @df7cb - this means that we need to try building and running pgloader on 32-bit Lisp implementations to iron out all the kinks that we're getting. Or perhaps add a 32-bit target to Travis builds and try to keep it running.

We'll also need to get that backtrace from @ateiss for sure to pinpoint this particular issue.

**Comment by dimitri** [comment]
The `Failed to start the monitor thread` error was specific to the SBCL/CCL threading runtime in v3. **pgloader v4 uses Java virtual threads** (JDK 21 Project Loom) with no Common Lisp thread monitor. This error cannot occur in v4.

---
This issue is fixed or eliminated by the **pgloader v4** rewrite (PR #1705). Download the latest v4 JAR:
```
curl -L -o pgloader.jar https://github.com/dimitri/pgloader/releases/download/v4-dev/pgloader.jar
java -jar pgloader.jar --version
```
